frontend/src/lib/components/IntervalFilter/intervalFilterLogic.ts

```diff
@@ -138,6 +147,114 @@ export const intervalFilterLogic = kea<intervalFilterLogicType>({
+                    break
+                }
+            }
+
+            const enabledIntervals: Intervals = { ...intervals }
+
+            if (activeUsersMath) {
+                // Disallow grouping by hour for WAUs/MAUs as it's an expensive query that produces a view that's not useful for users
+                enabledIntervals.hour = {
+                    ...enabledIntervals.hour,
+                    disabledReason:
+                        'Grouping by hour is not supported on insights with weekly or monthly active users series.',
+                }
+
+                // Disallow grouping by month for WAUs as the resulting view is misleading to users
+                if (activeUsersMath === BaseMathType.WeeklyActiveUsers) {
+                    enabledIntervals.month = {
+                        ...enabledIntervals.month,
+                        disabledReason:
+                            'Grouping by month is not supported on insights with weekly active users series.',
+                    }
+                }
+            }
+
+            actions.setEnabledIntervals(enabledIntervals)
+
+            // If the user just flipped an event action to use WAUs/MAUs math and their
+            // current interval is unsupported by the math type, switch their interval
+            // to an appropriate allowed interval and inform them of the change via a toast
+            if (
+                activeUsersMath &&
+                (values.querySource as TrendsQuery)?.interval &&
+                enabledIntervals[(values.querySource as TrendsQuery).interval as IntervalType].disabledReason
+            ) {
+                if (values.interval === 'hour') {
+                    lemonToast.info(
+                        `Switched to grouping by day, because "${BASE_MATH_DEFINITIONS[activeUsersMath].name}" does not support grouping by ${values.interval}.`
+                    )
+                    actions.updateQuerySource({ interval: 'day' } as Partial<InsightQueryNode>)
+                } else {
+                    lemonToast.info(
+                        `Switched to grouping by week, because "${BASE_MATH_DEFINITIONS[activeUsersMath].name}" does not support grouping by ${values.interval}.`
+                    )
+                    actions.updateQuerySource({ interval: 'week' } as Partial<InsightQueryNode>)
+                }
+                return
+            }
+
+            if (
+                !date_from ||
+                (objectsEqual(date_from, previousDateRange.date_from) &&
+                    objectsEqual(date_to, previousDateRange.date_to))
+            ) {
+                return
+            }
+
+            // automatically set an interval for fixed date ranges
+            if (
+                date_from &&
+                date_to &&
+                dayjs(querySource.dateRange?.date_from).isValid() &&
+                dayjs(querySource.dateRange?.date_to).isValid()
```

Comment:
Quick Q as I just noticed: Why is this `dateRange.date_from` and not `dateRange.from` if we're already changing the data shape?

Reply:
Good Q! And no real reason other than I wanted to keep it simple when I started working on data exploration and haven't gotten to improving the schema yet. I'm hoping we can version the schema and iterate on this.